### PR TITLE
Fix #389 and #422

### LINKED ISSUE #389: Non-interactive usage through flags
if you know the choice/anime_id and episode:
`ani-cli  boruto -a 3 -e 2`

searches for boruto looks for (english sub) and chooses episode 2 if exists then plays into mpv 

NO prompt asking to select episode!
Goes right to playing!

### PR DESCRIPTION
**Pull request by 71zenith**
It cannot accept a range of episodes at this point and settings subs is not supported cause I cannot find any thing in the script about that.

### CODE REVIEW

**Comment by 71zenith** [comment]
Made so that if both episode and anime is passed, it will non-interactively select the first result and start playing. Only thing left is merging which breaks the former.

File: ani-cli [hunk 1]
```diff
@@ -17,13 +17,14 @@ help_text () {
...
 	Usage:
-	  $0 [-v | -i] [-q <quality>] [-e <arguments>] [-d | -p <download_dir>] [<query>]
+	  $0 [-v | -i] [-q <quality>] [-e <arguments>] [-a <episode>] [-d | -p <download_dir>] [<query>]
 	  $0 [-v | -i] [-q <quality>] [-e <arguments>] -c
 	  $0 -h | -D | -U | -V
 
 	Options:
 	  -c continue watching anime from history
 	  -e pass arguments to the player/downloader
+	  -a specify episode to watch
```
Comment (ura43) [inline ani-cli:27]: - [ ] Add note that always first result is used, somewhere below the option list

File: ani-cli [hunk 2]
```diff
@@ -251,32 +252,35 @@ menu_line_strong() {
...
-			selection_id=$anime_id
-			break
-		fi
-		count=$((count+1))
-	done <<-EOF
-	$search_results
-	EOF
+	if [ -n "$ep_choice_to_start" ] && [ -n "$select_first" ]; then
+		selection_id="$(echo $search_results | sed 's/\ .*//' )"
+	elif [ -z "$ep_choice_to_start" ] || { [ -n "$ep_choice_to_start" ] && [ -z "$select_first" ]; }; then
```
Comment (ura43) [inline ani-cli:257]: - [ ] excessive use of || and && inside an if else directive is bad form

File: ani-cli [hunk 3]
```diff
@@ -307,11 +308,14 @@ episode_selection () {
...
-		is_half_ep=$(cut -c1-1 "$ep_choice_start")
-		if [ "$is_half_ep" = "h" ]
+		if [ -z "$ep_choice_to_start" ]; then
+			prompt "Choose episode" "[$first_ep_number-$last_ep_number]"
+			ep_choice_start=$REPLY
+			ep_choice_end=$REPLY2
+		else
+			ep_choice_start=$ep_choice_to_start && unset ep_choice_to_start
+		fi
+		if [ "$(echo "$ep_choice_start" | cut -c1-1)" = "h" ]
```
Comment (ura43) [inline ani-cli:318]: - [ ] Evaluate the cut command outside the if condition (SC2312)
- [ ] Unnescessary echo into cut. Just give the argument directly to cut

File: ani-cli [hunk 4]
```diff
@@ -251,32 +252,35 @@ menu_line_strong() {
...
-		if [ "$count" -eq "$choice" ]; then
-			selection_id=$anime_id
-			break
-		fi
-		count=$((count+1))
-	done <<-EOF
-	$search_results
-	EOF
+	if [ -n "$ep_choice_to_start" ] && [ -n "$select_first" ]; then
+		selection_id="$(echo $search_results | sed 's/\ .*//' )"
```
Comment (ura43) [inline ani-cli:256]: - [ ] Don't echo pipe into sed
- [ ] Is it really impossible to put $search_results in quotes?

**Review by ura43 — CHANGES_REQUESTED** [review]
- [ ] bump the version to 1.4.2

File: ani-cli (hunk 2 above)
Comment (71zenith) [reply ani-cli:257]: how do i put it in good form?

File: ani-cli (hunk 4 above)
Comment (71zenith) [reply ani-cli:256]: Cannot do either cause it would break.

File: ani-cli (hunk 3 above)
Comment (71zenith) [reply ani-cli:318]: Did it

File: ani-cli (hunk 2 above)
Comment (ura43) [reply ani-cli:260]: Eh its fine, just realized stuff like this is still common in the script. Thats a task for someone else during refactoring

**Review by ura43 — CHANGES_REQUESTED** [review]
Tested working successfully.
Please skip the menu when running non-interactively, leading to the script just launching mpv and then exiting again.
That way its truly non-interactive

**Comment by 71zenith** [comment]
Did it

File: ani-cli (hunk 4 above)
Comment (71zenith) [reply ani-cli]: Fixed it by using a different method

**Review by ura43 — APPROVED** [review]
Confirmed functional with `fooly-cooly ep 3`
I'm glad you got it shellcheck compliant

**Pull request merged**